### PR TITLE
Improve \cancel

### LINKED ISSUE #1080: bounding box is wrong for \cancel groups
strikethrough

I added
```
\usepackage{cancel}
\usepackage{ulem}
```
to test.tex in order to generate the image above.  The spacing around binary operators is better than it used to be.  It seems like the addition of the `\cancel` bar doesn't affect the bounding box of `\cancel` in the LaTeX rendering (green).  The bounding box of `\cancel` groups is essentially the same as the bounding box of the contents.  This is probably most noticeably in `{\cancel B}_1^2`.

### PR DESCRIPTION
**Pull request by ronkok**
Fixes #1080.

Improve `\cancel` in three ways:

1. Better lateral spacing.
2. Steeper `\cancel` lines on single characters.
3. Treat height and depth to match the `cancel` package behavior, not its documentation.

### CODE REVIEW

**Comment by ronkok** [comment]
New and improved.
cancel

**Comment by kevinbarabash** [comment]
That looks much better.  It'd be nice to see a visual diff against LaTeX's output.  Unfortunately my docker setup is broken at the moment.

**Comment by kevinbarabash** [comment]
Screenshots from circleci:
StrikeThrough (Chrome)
strikethrough-chrome

StrikeThroughColor (Chrome)
strikethroughcolor-chrome

StrikeThrough (Firefox)
strikethrough-firefox

StrikeThroughColor (Firefox)
strikethroughcolor-firefox

@ronkok do you have access to the artifacts on the circleci builds?

**Comment by edemaine** [comment]
Here are the screenshot diffs, courtesy of circleci:

strikethrough-chrome-diff

strikethroughcolor-chrome-diff

The vertical differences are caused by changes in the YAML test itself, but the horizontal changes are the main impact of this PR.

Here's a texcmp for the first screenshot (with this PR):

strikethrough

I'm pretty sure this is better.  Here's what the texcmp used to look like, which is almost identical to how the screenshot diff looks! (a good sign)

strikethrough 1

**Comment by kevinbarabash** [comment]
Are those diffs on circleci diffs against KaTeX's previous version or against LaTeX's rendering?

**Comment by edemaine** [comment]
CircleCI diffs (my top images) are diffs between KaTeX versions (before and after PR).

My bottom images are texcmp, which compare KaTeX vs. LaTeX.

**Comment by kevinbarabash** [comment]
The top row in the texcmp rendering is almost perfect.  Overall this is a big improvement!

**Comment by ronkok** [comment]
Theres a discrepancy in the gap between lines 2 and 3. Could that be due to the problem with `\sout`?

**Comment by edemaine** [comment]
All of the line spacing is off by a similar amount.  My sense is that we're not measuring lengths correctly yet.  `\scriptstyle` also seems to be a different font size than what LaTeX uses.  Neither of these are issues for this PR

File: src/katex.less [hunk 1]
```diff
@@ -538,15 +538,10 @@
...
-    }
-    .cancel-lap + .mord,
-    .cancel-lap + .mbin,
-    .cancel-lap + .msupsub {
-        margin-left: -0.2em;
-    }
+    .cancel-lap {
+        margin-left: -0.2em;    // \cancel does not affect horizontal spacing.
+        margin-right: -0.2em;   // Apply negative margin to correct for 0.2em padding
+    }                           // inside the \cancel group.
```
[the remark this answers is not in the document]
Comment (ronkok) [reply src/katex.less]: We need to keep the classes. Consider the `\cancel{2B}` below.  It gets the wide treatment, so the line endpoints extend 0.2em beyond the subject matter. I do this by adding 0.2em padding on both ends of the subject matter, then letting the line run from across the whole container span.

But that 0.2em is not supposed to affect the width of the element. So I take the 0.2em back with negative margin on the container span, and that margin is needed on both right and left side.

smallcancel

[the remark this answers is not in the document]
Comment (kevinbarabash) [reply src/katex.less]: That makes sense.

**Pull request merged**